store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -97,13 +106,25 @@ private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> oper
 
     }
 
-    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation operation, final FederatedStore store) throws OperationException {
+    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation<INPUT, OUTPUT> operation, final FederatedStore store) throws OperationException {
         try {
             Object rtn = null;
 
             final BiFunction mergeFunction = getMergeFunction(operation, store, context, isEmpty(resultsFromAllGraphs));
+            final List<GraphSerialisable> graphs = getGraphs(operation, context, store);
+
+            // If default merging and only have one graph or no common groups then just return the current results
+            if (!graphs.isEmpty()
```

Comment:
Do we really need to check if `graphs` is empty? I _believe_ that if there are no graph then this method doesn't get run.

Reply:
I thought the same but lots of the federation unit tests fail without this check in as it seems lots of the tests don't create a graph but still run the merge function.

Reply:
Ok, when used normally I think it doesn't get run. Not ideal to have this to work with tests, could raise an issue for that. This can stay for this PR.

Reply:
Agreed I think its more an edge case with the testing.

Reply:
I think if you have an empty FederatedStore this might get run?

Reply:
I thought that with an empty store there was some logic to stop. Possibly in a similar place to code which logs the message about empty schemas. In any case this check is required so I'll resolve this.